lib/ddtrace/transport/traces.rb

```diff
@@ -92,6 +94,13 @@ module Encoder
         module_function
 
         def encode_trace(encoder, trace)
+          # Format the trace for transport
+          TraceFormatter.format!(trace)
+
+          # Make the trace serializable
+          trace = SerializableTrace.new(trace)
```

Comment:
I tend to find it slightly confusing when the type of an object is changed with an assignment back to the same local variable name.

Could we rename the new object, so we know that it is not the same type as the provided `trace` anymore? `serializable_trace`, for example, is fine, but it could be another name if you wish.

Reply:
Sure.